package.json

```diff
@@ -1,5 +1,14 @@
 {
     "devDependencies": {
+        "@types/mkdirp": "^0.5.1",
+        "@types/node": "^8.0.28",
+        "solc": "^0.4.16",
         "solium": "github:AugurProject/solium"
+    },
+    "dependencies": {
+        "fs": "0.0.1-security",
+        "mkdir-recursive": "^0.3.0",
+        "mkdirp": "^0.5.1",
+        "recursive-readdir": "^2.2.1"
```

Comment:
We should pick whether to do all `dependencies` or all `devDependencies`.  For this project, where nothing is "deployed" like a web server or website, I don't believe it matters which we use.

Reply:
I suppose we can just stick with devDependencies since it doesn't seem like it will make a difference.  I also noticed I'm including `mkdir-recursive`, which I don't use, so I'll remove it.

I haven't used node much before (my JS experience is primarily just on the front-end) -- do we actually need to be including `fs` here?  Isn't it a default module like `path`?  Also, can I just use 
```
"mkdirp": "^0.5.1",
```
and get rid of:
```
"@types/mkdirp": "^0.5.1",
```
What is the difference?

Reply:
`"mkdirp": "^0.5.1",` is Node's way of saying the project depends on the `mkdirp` package at runtime.

`"@types/mkdirp": "^0.5.1",` is TypeScript's way of pulling in definitions for the `mkdirp` package so that the compiler doesn't error when you try to use it at compile time.  When working with a JavaScript package like `mkdirp`, you either need the package + one or more `.d.ts` files _OR_ you need the package + `@types/<package>`.  Basically, `@types/<package>` just pulls in `.d.ts` files that someone else created for some JavaScript package.

If the package is natively TypeScript or it includes its own `.d.ts` files then you need neither the @types nor a `.d.ts` file.  Most packages on NPM are JS without d.ts files, so you'll _usually_ need one of the two.

----

I don't believe you need to depend on `fs`.  I think that is actually just a placeholder package on NPM to prevent attackers from replacing node's `fs` library with something malicious and pulling it down as a dependency doesn't do anything.  You can safely remove that line.

Reply:
Ok, thanks for the clarification.  It sounds like I need `"mkdirp": "^0.5.1",` and `"@types/mkdirp": "^0.5.1",`.